### DISCUSSION TITLE
Fail2ban doesn't start (Ubuntu 22.04 & Debian 12)

### DISCUSSION THREAD

**Discussion by WhiteTiger-IT** [fail2ban/fail2ban#3680] [General]
On Ubuntu server 22.04.3 Fail2Ban does not start with
sudo systemctl enable --now fail2ban
sudo systemctl status fail2ban

The error seems to be related to bantime, but I tried with 60m, with 1h and with 1d.

Where do I find the logs?
I can't find any updated data in /var/log/fail2ban.log.

This is what it appears in the terminal

`$ systemctl status fail2ban
× fail2ban.service - Fail2Ban Service
     Loaded: loaded (/lib/systemd/system/fail2ban.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Wed 2024-02-21 02:59:53 CET; 3s ago
       Docs: man:fail2ban(1)
    Process: 7763 ExecStart=/usr/bin/fail2ban-server -xf start (code=exited, status=255/EXCEPTION)
   Main PID: 7763 (code=exited, status=255/EXCEPTION)
        CPU: 231ms

Feb 21 02:59:53 pc-sys.mydomain.tld systemd[1]: Started Fail2Ban Service.
Feb 21 02:59:53 pc-sys.mydomain.tld fail2ban-server[7763]: 2024-02-21 02:59:53,528 fail2ban                [7763]: ERROR   Failed during configuration: File contains no section headers.
Feb 21 02:59:53 pc-sys.mydomain.tld fail2ban-server[7763]: file: '/etc/fail2ban/jail.conf', line: 21
Feb 21 02:59:53 pc-sys.mydomain.tld fail2ban-server[7763]: 'bantime = 60m\n'
Feb 21 02:59:53 pc-sys.mydomain.tld fail2ban-server[7763]: 2024-02-21 02:59:53,532 fail2ban                [7763]: ERROR   Async configuration of server failed
Feb 21 02:59:53 pc-sys.mydomain.tld systemd[1]: fail2ban.service: Main process exited, code=exited, status=255/EXCEPTION
Feb 21 02:59:53 pc-sys.mydomain.tld systemd[1]: fail2ban.service: Failed with result 'exit-code'.

`

**Comment by WhiteTiger-IT** [comment]
I did a second installation on Debian 12, clean (there is only Webmin).
It doesn't start on this either, but with a different error referring to IPV6.
There is no IPV6 or even DHCP-V6 on my home network.
However I disabled IPV6 on the network card by editing the file  **/etc/sysctl.conf** adding at the bottom
`
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.lo.disable_ipv6 = 1
net.ipv6.conf.tun0.disable_ipv6 = 1
`


`× fail2ban.service - Fail2Ban Service
     Loaded: loaded (/lib/systemd/system/fail2ban.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Wed 2024-02-21 04:05:07 CET; 5h 24min ago
   Duration: 3.518s
       Docs: man:fail2ban(1)
    Process: 663 ExecStart=/usr/bin/fail2ban-server -xf start (code=exited, status=255/EXCEPTION)
   Main PID: 663 (code=exited, status=255/EXCEPTION)
        CPU: 318ms

feb 21 04:05:04 ls1 systemd[1]: Started fail2ban.service - Fail2Ban Service.
feb 21 04:05:07 ls1 fail2ban-server[663]: 2024-02-21 04:05:07,809 fail2ban.configreader   [663]: WARNING 'allowipv6' not defined in 'Definition'. Using default one: 'auto'
feb 21 04:05:07 ls1 fail2ban-server[663]: 2024-02-21 04:05:07,837 fail2ban                [663]: ERROR   Failed during configuration: Have not found any log file for sshd jail
feb 21 04:05:07 ls1 fail2ban-server[663]: 2024-02-21 04:05:07,844 fail2ban                [663]: ERROR   Async configuration of server failed
feb 21 04:05:07 ls1 systemd[1]: fail2ban.service: Main process exited, code=exited, status=255/EXCEPTION
feb 21 04:05:07 ls1 systemd[1]: fail2ban.service: Failed with result 'exit-code'.
`

**Reply by sebres** [reply]
> `ERROR Failed during configuration: Have not found any log file for sshd jail`

See https://github.com/fail2ban/fail2ban/issues/3292#issuecomment-1142503461

**Comment by sebres** [comment]
Error `File contains no section headers.` says what it says - you need at least one section `[...]` in the config-file, either `[DEFAULT]` for default, or for some jail e. g. `[sshd]` for sshd jail.

**Comment by WhiteTiger-IT** [comment]
> > `ERROR Failed during configuration: Have not found any log file for sshd jail`
> 
> See [#3292 (comment)](https://github.com/fail2ban/fail2ban/issues/3292#issuecomment-1142503461)

I had already read them yesterday, but I confess that I didn't understand either what the problem is and how to solve it.

**Reply by sebres** [reply]
Hmm... what exactly is so complex to understand there?

The error means that for the jail sshd fail2ban cannot find a log-file specified by the logpath (`/var/log/auth.log`).
This can mostly happen, if the path is different or, as in Debian 12, because sshd logs into systemd journal and not into the log-file. 
To force journal monitoring this jail must get `backend = systemd` setting, what this comment says.

Thus open `/etc/fail2ban/jail.local` to edit and add to `[sshd]` section (where you also enable it) a new line `backend = systemd`.
That's it.

**Reply by WhiteTiger-IT** [reply]
You were very clear. I thank you. Now that error is no longer there.
I imagine this should also be done for the other enabled jails, right?

**Reply by sebres** [reply]
> I imagine this should also be done for the other enabled jails, right?

Only if the services corresponding the jails logging to journal (check journalctl or their log-file).
The simplest method can be also the check with `fail2ban-regex`:
```
fail2ban-regex systemd-journal "$jail"
fail2ban-regex /path/to/log "$jail"
```
(if first shows some found messages, it is rather journal, otherwise the second must find something)

**Reply by duzhuoshanwai** [reply]
> Hmm... what exactly is so complex to understand there?嗯......到底是什么让人如此难以理解？
> 
> The error means that for the jail sshd fail2ban cannot find a log-file specified by the logpath (`/var/log/auth.log`).该错误表示 jail sshd fail2ban 无法找到 logpath ( `/var/log/auth.log` ) 指定的日志文件。 This can mostly happen, if the path is different or, as in Debian 12, because sshd logs into systemd journal and not into the log-file.如果路径不同，或者在 Debian 12 中，由于 sshd 会将日志记录到 systemd 日志中，而不是日志文件中，就会出现这种情况。 To force journal monitoring this jail must get `backend = systemd` setting, what this comment says.要强制日志监控，这个监狱必须设置 `backend = systemd` ，这就是这条评论所说的。
> 
> Thus open `/etc/fail2ban/jail.local` to edit and add to `[sshd]` section (where you also enable it) a new line `backend = systemd`.因此，打开 `/etc/fail2ban/jail.local` 进行编辑，并在 `[sshd]` 部分（也是启用它的地方）添加新行 `backend = systemd` 。 That's it. 就是这样。

thats work for me
Debian 12.8
